dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/ImageStoreUtil.java

```diff
@@ -0,0 +1,42 @@
+package com.salesforce.dockerfileimageupdate.utils;
+
+import com.salesforce.dockerfileimageupdate.storage.ImageStoreType;
+import com.salesforce.dockerfileimageupdate.storage.ImageTagStore;
+import com.salesforce.dockerfileimageupdate.storage.S3BackedImageTagStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * ImageStore is the main entity we'll be using to initialize the image tag store
+ * @author amukhopadhyay
+ */
+public class ImageStoreUtil {
```

Comment:
This looks like a utility class as it doesn't store instance state.  In that case, make all of the methods static, make this class final and add the following constructor:

```
private ImageStoreUtil() {}
```
This prevents the creation of instances of this class.

Reply:
"This is being initialized in the All and Parent class."  With what?  Basically if you are doing the following in Java
```
ImageStoreUtil util = new ImageStoreUtil()
util.initializeImageTagStore();
```
And `ImageStoreUtil` maintains no instance (i.e non-static) variables then this is equivalent but less resource intensive
```
ImageStoreUtil.initializeImageTagStore();
```
You just need to make initializeImageTagStore static because its a pure function.

Reply:
@cjneasbi isn't it going to be hard to mock out static methods? We generally follow the object initialization pattern so we can mock the while utility out when not unit testing the util itself.

Reply:
In cases where you are trying to call a static method replace this with a Lambda parameter and that static method call is no longer static.

```
public SomeClass{
public static void doAThing();
}

...

public myMethod(){
  SomeClass.doAThing()
}
```

becomes

```
public SomeClass{
public static void doAThing();
}

...

public myMethod(){
  myMethod(SomeClass::doAThing)
}

myMethod(Supplier<Void> thingDoer){
  thingDoer.get()
}

``` 

Then you test `myMethod(Supplier<Void> thingDoer)`

```
@Test
public void myTest()
{
  myMethod(() -> null)
}
```
 
Now you don't need Mockito to mock your static logic